src/game/AI/ScriptDevAI/scripts/world/scourge_invasion.cpp

```diff
@@ -886,7 +974,7 @@ struct npc_cultist_engineer : public ScriptedAI
         int32 eT = eventType;
         if (eT == 7166 && miscValue == 0)
         {
-            if (Player* player = dynamic_cast<Player*>(invoker))
+            if (auto* player = dynamic_cast<Player*>(invoker))
```

Comment:
I advise against using auto here, simple types like this do not save much time typing vs the readability. With dynamic cast its fairly trivial but what if its a complex function.